Dockerfile

```diff
@@ -18,56 +18,38 @@
 FROM node:18-bookworm as build
 RUN node --version && npm --version
 
-# Prepare the folder to enable non-root, otherwise npm will refuse to run the postinstall
-RUN mkdir /vault
-RUN chown node:node /vault
-USER node
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 # Can be a tag, release, but prefer a commit hash because it's not changeable
 # https://github.com/bitwarden/clients/commit/${VAULT_VERSION}
 #
 # Using https://github.com/bitwarden/clients/releases/tag/web-v2024.5.1
 ARG VAULT_VERSION=9823f69c9d17e2d94de1cc005e01202dd95f0647
+ENV VAULT_VERSION=$VAULT_VERSION
+ENV VAULT_FOLDER=bw_clients
 
-WORKDIR /vault
-RUN git -c init.defaultBranch=main init && \
-    git remote add origin https://github.com/bitwarden/clients.git && \
-    git fetch --depth 1 origin "${VAULT_VERSION}" && \
-    git -c advice.detachedHead=false checkout FETCH_HEAD
+RUN mkdir /bw_web_builds
+WORKDIR /bw_web_builds
 
-COPY --chown=node:node patches /patches
-COPY --chown=node:node resources /resources
-COPY --chown=node:node scripts/apply_patches.sh /apply_patches.sh
+COPY patches ./patches
+COPY resources ./resources
+COPY scripts ./scripts
 
-RUN bash /apply_patches.sh
-
-# Build
-RUN npm ci
-
-# Switch to the web apps folder
-WORKDIR /vault/apps/web
-
-RUN npm run dist:oss:selfhost
-
-RUN printf '{"version":"%s"}' \
-      $(git -c 'versionsort.suffix=-' ls-remote --tags --refs --sort='v:refname' https://github.com/dani-garcia/bw_web_builds.git 'v*' | tail -n1 | grep -Eo '[^\/v]*$') \
-      > build/vw-version.json
-
-# Delete debugging map files, optional
-# RUN find build -name "*.map" -delete
-
-# Prepare the final archives
-RUN mv build web-vault
-RUN tar -czvf "bw_web_vault.tar.gz" web-vault --owner=0 --group=0
+RUN ./scripts/checkout_web_vault.sh; \
+	./scripts/patch_web_vault.sh; \
+	./scripts/build_web_vault.sh; \
+	mv $VAULT_FOLDER/apps/web/build ./web-vault
 
+RUN tar -czvf "bw_web_vault.tar.gz" ./web-vault
```

Comment:
I would suggest to still keep the `owner` and `group` arguments for tar. Main reason is that people could running this as a different user maybe. And this at least ensures all files will have the correct user and group within the whole file

Reply:
I believe the `0` was needed to set it to `root` since it was running as `node`.

Reply:
People can still run it as non privileged use like with podman which might cause it to have a different user and group as expected.